test/datatype.test.js

```diff
@@ -220,12 +220,12 @@ describe('datatypes', function() {
 
     it('should convert property value undefined to null', function(done) {
       var EXPECTED = {desc: null, extra: null};
+      var data = {desc: undefined, extra: undefined};
       if (isStrict) {
         // SQL-based connectors don't support dynamic properties
         delete EXPECTED.extra;
+        delete data.extra;
```

Comment:
I'm not sure what to do with this test case, so the option of `{ persistUndefinedAsNull: true }` disables the discarding of undefined property, while `strict mode` will return validation error. 

in this case deleting `data.extra` is needed ( and i think is correct behavior? )
while the test will still check for the defined property (`desc`) which fulfills the original intent of the test